src/Stratis.Bitcoin.Features.Consensus.Tests/TimeSyncBehaviorTest.cs

```diff
@@ -28,173 +28,283 @@ public class TimeSyncBehaviorTest
+            for (int i = TimeSyncBehaviorState.MaxInboundSamples; i < TimeSyncBehaviorState.MaxInboundSamples*2; i++)
+            {
+                Assert.True(state.AddTimeData(new IPAddress(i*2), TimeSpan.FromSeconds(-1000), true), $"index: {i}");
+            }
+
+            diff =  dateTimeProvider.GetAdjustedTime() -  dateTimeProvider.GetUtcNow();
+            Assert.True(Math.Abs(diff.TotalMilliseconds - (-250*1000)) < TimeEpsilonMs, $"should be -250ms because the new inbound replace the old and swing the median closer to the new -1000ms values, but still in outbound. Actual: {diff.TotalMilliseconds}ms");
+        }
+         
+        /// <summary>
+        /// Sets up the system under test
+        /// </summary>
+        /// <param name="dateTimeProvider"></param>
+        /// <returns></returns>
+        private TimeSyncBehaviorState CreateTimeSyncBehaviorState(DateTimeProvider dateTimeProvider)
+        {
+            var lifetime = new NodeLifetime();
+            var loggerFactory = new LoggerFactory();
+            var asyncLoopFactory = new AsyncLoopFactory(loggerFactory);
+            var state = new TimeSyncBehaviorState(dateTimeProvider, lifetime, asyncLoopFactory, loggerFactory, Network.Main);
+            return state;
+        }
+    }
+
+    /// <summary>
+    /// Representation of a Sample for data driven tests.
+    /// </summary>
+    public class TestSample
+    {
+        /// <summary>
+        /// Input isInbound flag, with false meaning outbound.
+        /// </summary>
+        public bool IsInbound { get; }
 
-            adjustedTime = dateTimeProvider.GetAdjustedTime();
-            normalTime = dateTimeProvider.GetUtcNow();
-            diff = adjustedTime - normalTime;
-            expectedDiffMs = allSamples.Median();
-            Assert.True(Math.Abs(diff.TotalMilliseconds) < Math.Abs(expectedDiffMs) + TimeEpsilonMs);
+        /// <summary>
+        /// Expectation of whether the sample is used or not.
+        /// </summary>
+        public bool ExpectedIsUsed { get; }
+        
+        
+        /// <summary>
+        /// >Expectation of whether the warning has been set to on.
+        /// </summary>
+        public bool ExpectedIsWarningOn { get; }
+        
+        /// <summary>
+        /// 
+        /// </summary>
+        public bool ExpectedIsSyncOff { get; }
+        
+        /// <summary>
+        /// Expectation that the time offset will be set to this value.
+        /// </summary>
+        public int ExpectedTimeOffsetMs { get; }
+        
+        /// <summary>
+        /// The actual inout time offset from the sample.
+        /// </summary>
+        public TimeSpan InputTimeOffset { get; }
+        
+        /// <summary>
+        /// The IP Address of the sample.
+        /// </summary>
+        public IPAddress PeerIpAddress { get; }
+
+        /// <summary>
+        /// Private constructor, forcing the factory methods Inbound and Outbound to be used.
```

Comment:
If you want to reference a method or something, always use `<see>`

Reply:
seems ridiculously unnecessary int this situation.

Reply:
It is just good practice to prevent having invalid comments after refactoring the name. If used automated tool for the refactoring, which is probably what you always use, then if you use `see`, it will automatically update your comment as well and you don't need to do that manually and forget about it